applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py

```diff
@@ -234,6 +252,24 @@ def InitializeSolutionStep(self):
     def FinalizeSolutionStep(self):
         KratosMultiphysics.Logger.PrintInfo(self.__class__.__name__, "Mass and momentum conservation equations are solved.")
 
+        if (self._levelset_splitting):
+            self.main_model_part.ProcessInfo.SetValue(KratosMultiphysics.DELTA_TIME_FACTOR, (1.0 - self._levelset_dt_factor))
+            # Perform the level-set convection to complete the solution step
+            if self._bfecc_convection:
+                self._GetLevelSetConvectionProcess().CopyScalarVarToPreviousTimeStep(
+                    self.main_model_part,
+                    KratosMultiphysics.DISTANCE)
+                self._GetLevelsetGradientProcess().Execute() #Level-set gradient is needed for the limiter
+                self._GetLevelSetConvectionProcess().BFECCconvect(
+                    self.main_model_part,
+                    KratosMultiphysics.DISTANCE,
+                    KratosMultiphysics.VELOCITY,
+                    self.settings["bfecc_number_substeps"].GetInt())
+            else:
+                if (self._eulerian_error_compensation):
+                    self._GetLevelsetGradientProcess().Execute() #Level-set gradient is needed for the limiter
+                self._GetLevelSetConvectionProcess().Execute()
```

Comment:
Isn't this copied and pasted from the piece of code in the `InitializeSolutionStep`? If so it is, it needs to be encapsulated in an auxiliary "private" function.

Reply:
Done.